utest/src-0/utest/TestBuilder.scala

```diff
@@ -0,0 +1,130 @@
+
+import utest.framework.{TestCallTree, Tree => UTree, TestPath }
+
+
+class TestBuilder(given QuoteContext) extends TestBuilderExtractors {
+  import qc.tasty.{ Tree => TasTree, given, _ }
+
+  def buildTestsTrees(tests: List[Apply], path: Seq[String]): (List[Expr[UTree[String]]], List[Expr[TestCallTree]]) =
+    if tests.isEmpty then Nil -> Nil else tests.zipWithIndex.foldLeft((List.empty[Expr[UTree[String]]], List.empty[Expr[TestCallTree]])) {
+      case ((namesForest, callsForest), (nextTest, id)) =>
+        val (name, body) = processTest(nextTest, path, id)
+        (namesForest :+ name, callsForest :+ body)
+    }
+
+  def TestCallTreeExpr(nestedBodyTrees: List[Expr[TestCallTree]], setupStats: List[Statement]): Expr[TestCallTree] = '{TestCallTree { ${(
+    if (nestedBodyTrees.nonEmpty)
+      Block(setupStats, '{Right(${Expr.ofList(nestedBodyTrees)}.toIndexedSeq)}.unseal)
+    else
+      Block(setupStats.dropRight(1), '{Left(${setupStats.takeRight(1).head.asInstanceOf[Term].seal})}.unseal)
+    ).seal.cast[Either[Any, IndexedSeq[TestCallTree]]]
+  }}}
+
+
+  def processTest(test: Apply, pathOld: Seq[String], index: Int): (Expr[UTree[String]], Expr[TestCallTree]) = test match {
+    case Test(nameOpt, nestedTests, setupStatsRaw) =>
+      val name = nameOpt.getOrElse(index.toString)
+      val path = pathOld :+ name
+      val (nestedNameTrees, nestedBodyTrees) = buildTestsTrees(nestedTests, path)
+
+      object testPathMap extends TreeMap {
+        override def transformTerm(t: Term)(implicit ctx: Context): Term =
+          t.tpe.widen match {
+            case _: MethodType | _: PolyType => super.transformTerm(t)
+            case _ => t.seal match {
+              case '{TestPath.synthetic} => '{TestPath(${path.toExpr})}.unseal
+              case _ => super.transformTerm(t)
+            }
+          }
+      }
+
+      val setupStats = testPathMap.transformStats(setupStatsRaw)
+
+      val names: Expr[UTree[String]] = '{UTree[String](${name.toExpr}, ${Expr.ofList(nestedNameTrees)}: _*)}
+      val bodies: Expr[TestCallTree] = TestCallTreeExpr(nestedBodyTrees, setupStats)
+
+      (names, bodies)
+  }
+
+  def processTests(body: Term): Expr[Tests] =
+    @annotation.tailrec def unwrap(t: Term): Term = t match
+      case Inlined(_, _, x) => unwrap(x)
+      case x => x
+
+    unwrap(body) match {
+      case Stats(tests, setupStats) =>
+        val (nestedNameTrees, nestedBodyTrees) = buildTestsTrees(tests, Vector())
+        '{Tests(UTree[String](
+            "", ${Expr.ofList(nestedNameTrees)}: _*)
+          , ${TestCallTreeExpr(nestedBodyTrees, setupStats)})}
+    }
+}
+
+trait TestBuilderExtractors(given val qc: QuoteContext) {
+  import qc.tasty.{ given, _ }
+
+  object TestMethod {
+    def (strExpr: Expr[String]) exec (given v: ValueOfExpr[String]): Option[String] = v(strExpr)
+
+    def unapply(tree: Tree): Option[(Option[String], Tree)] =
+      Option(tree).collect { case IsTerm(tree) => tree.seal }.collect {
```

Comment:
Are IsXYZ extractors already deprecated in this version? If so, should update them to `tree: Tree`.